app/src/main/java/com/hedvig/app/feature/swedishbankid/sign/SwedishBankIdSignDialog.kt

```diff
@@ -0,0 +1,177 @@
+import com.hedvig.app.util.extensions.viewLifecycleScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+
+class SwedishBankIdSignDialog : DialogFragment() {
+    private val model: SwedishBankIdSignViewModel by viewModel {
+        parametersOf(
+            requireArguments().getString(AUTO_START_TOKEN)
+                ?: throw IllegalArgumentException(
+                    "Programmer error: Missing AUTO_START_TOKEN in ${this.javaClass.name}"
+                ),
+            requireArguments().getStringArrayList(QUOTE_IDS)
+                ?: throw IllegalArgumentException(
+                    "Programmer error: Missing QUOTE_IDS in ${this.javaClass.name}"
+                )
+        )
+    }
+    private val marketManager: MarketManager by inject()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnLifecycleDestroyed(viewLifecycleOwner))
+        setContent {
+            val viewState by model.viewState.collectAsState()
+            HedvigTheme {
+                SwedishBankIdSignDialog(
+                    text = stringResource(
+                        when (viewState) {
+                            is SwedishBankIdSignViewModel.ViewState.StartClient -> R.string.SIGN_START_BANKID
+                            SwedishBankIdSignViewModel.ViewState.Cancelled -> R.string.SIGN_CANCELED
+                            SwedishBankIdSignViewModel.ViewState.InProgress -> R.string.SIGN_IN_PROGRESS
+                            SwedishBankIdSignViewModel.ViewState.Success -> R.string.SIGN_SUCCESSFUL
+                            SwedishBankIdSignViewModel.ViewState.Error -> R.string.SIGN_FAILED_REASON_UNKNOWN
+                        }
+                    )
+                )
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        model
+            .events
+            .flowWithLifecycle(viewLifecycle)
+            .onEach { event ->
+                when (event) {
+                    is SwedishBankIdSignViewModel.Event.StartBankID -> {
+                        val bankIdUri = bankIdUri(event.autoStartToken)
+                        if (requireActivity().canOpenUri(bankIdUri)) {
+                            startActivity(
+                                Intent(
+                                    Intent.ACTION_VIEW, bankIdUri
+                                )
+                            )
+                        }
+                    }
+                    SwedishBankIdSignViewModel.Event.StartDirectDebit -> {
+                        marketManager
+                            .market
```

Comment:
Should we check that the current market is SE here since this is for SwedishBankId? Perhaps that is too defensive :)

Reply:
Good question! What would the consequences of being in the wrong market be? 🤔

Reply:
I suppose we only want to start `TrustlyConnectPayinActivity` here, and if we (for some reason) dont have a market this will just do nothing

Reply:
I kind of like the 'looser' coupling we get from using `Market::connectPayin` 😉 if we change what `Activity` we launch for connecting payin for the Swedish market, it's kind of good to have it in one place I think 🤔 

Personally, I feel like the consequences of having the wrong market will be, before this happens, that the member simply will not be able to sign (as a member outside this market will not have a Swedish BankID). What do you think? 🙂

Reply:
Yeah I agree, fair points!